### DISCUSSION TITLE
LIST of smartwatches as collectors connected successfully to juggluco

### DISCUSSION THREAD

**Discussion by Npavkov** [j-kaltes/Juggluco#74] [General]
From reading the discussions and issues on this juggluco page, there seems to be MANY individuals (including myself) that are VERY INTERESTED in connecting ANY smartwatch up to use AS A COLLECTOR. and send data to phone. Can a discussion be started that shares ONLY the models of smartwatches that are CONFIRMED to work as collectors with Juggluco.?

Watch Model, Sensor model, version of juggluco, and optionally, the process used to configure the smartwatch to work as a collector.

I know myself and my others would be very interested in this information......Thank You.

**Comment by jrsmile** [comment]
hi, i might give it a try.
i was successful in installing juggluco 5.1.5 on my galaxy watch 6 classic lte.
together with the eu version of the libre 3.
the phone to watch mode works good and i have no cutouts or battery optimisation based app crashes.
if however is switch to direct sensor connection only every 3rd or 5th sensor data interval ends up beeing recogniced.
as if the watch activates bluetooth only every 2 minutes or so. no combination of wakelocks or android13 tweaks etc. makes a difference here. But, it is usable and consistent if you dont need a value every minute.
the big problem i had and why i returned back to phone mode is that the battery is drained by juggluco excepionaly fast, over 33% of the overall battery budget goes to the app in background. expect less then 8 hours if juggluco is active.

another problem i had was the lack of the libreview upload feature in the wear os version.
i can imagine if it would be available for wear os the battery drain because of active lte module and or wifi would be even more problematic, but it was one of the main reasons for me to consider using a smartwatch for sensor collection.

i havent tested if the watch can start a libre 3 via nfc or if novopen 6 works on the watch to trigger insulin data upload.

i hope this helps.
Regards,
J.

**Reply by j-kaltes** [reply]
Do you mean with "_if however is switch to direct sensor connection only every 3rd or 5th sensor data interval ends up beeing recogniced_"? That you receive values every 3 or 5 minutes instead of every minute in the watch, but that there are no values missing for long periods? This is really unusual. Watch 6 is nearly identical to Watch 4, except for _Bluetooth 5.3, A2DP, LE_, while for Watch 4 it is, _Bluetooth 5.0, A2DP, LE_.
Also the Libre 3 sensors are not very old, so you would expect it work better, not worse with a newer Bluetooth version.
When one switches to a direct connection of the watch it takes 3 to 5 minutes before values are received by the watch, but thereafter they are received every minute, as least on Watch 4. When Watch 4 has connection problems they often last much longer often hours, not only 3 or 5 minutes.
If you still remember when you tried out directly connecting the phone to the Watch, you can export the Stream values (Left middle menu->Export->Stream) and E-Mail (jaapkorthalsaltes@gmail.com) them to me, so I can see what happes. Did you look at Left menu->Sensor on the watch when you had to wait more than a minute for the next value?
Also your description of the battery drainage is remarkable. Normally directly or indirectly connecting the watch with the sensor doesn't make a difference. And where do you see that Juggluco uses 33% of the battery budget. When I look in the Wearable app under Watch settings->Battery, Juggluco isn't shown under the apps listen there. Listed are: Display 2.1%, Samsung Heath 1.2%, Health service 0.6%, One Ui Watch Home 0.3%, Google Play Store 0.2%, Watch Faces 0.2%.
You have there at the top Juggluco 33.3% ??????
![battery](https://github.com/j-kaltes/Juggluco/assets/72013079/851bf498-6ee5-477d-97e5-81069627900f)

Under Watch settings->Apps->Juggluco is mentioned "_Battery 0% used since last fully charged_".
![appinfo](https://github.com/j-kaltes/Juggluco/assets/72013079/b5fb21e4-efe1-4060-8d5a-f8ed4d36a746)

NFC can't be used any application under WearOS, it is monopolized by Google Pay.

**Comment by jrsmile** [comment]
> Do you mean with "_if however is switch to direct sensor connection only every 3rd or 5th sensor data interval ends up beeing recogniced_"? That you receive values every 3 or 5 minutes instead of every minute in the watch, but that there are no values missing for long periods? This is really unusual. Watch 6 is nearly identical to Watch 4, except for _Bluetooth 5.3, A2DP, LE_, while for Watch 4 it is, _Bluetooth 5.0, A2DP, LE_. Also the Libre 3 sensors are not very old, so you would expect it work better, not worse with a newer Bluetooth version. When one switches to a direct connection of the watch it takes 3 to 5 minutes before values are received by the watch, but thereafter they are received every minute, as least on Watch 4. When Watch 4 has connection problems they often last much longer often hours, not only 3 or 5 minutes. 

yes you got this right, my thoughts are that they "enhanced" their powersaving strategy or that it might be an interference with other bluetooth devices connected to the watch (wireless earbuds) that prevent the watch from getting data every minute.

>If you still remember when you tried out directly connecting the phone to the Watch, you can export the Stream values (Left middle menu->Export->Stream) and E-Mail ([jaapkorthalsaltes@gmail.com](mailto:jaapkorthalsaltes@gmail.com)) them to me, so I can see what happes. Did you look at Left menu->Sensor on the watch when you had to wait more than a minute for the next value? 

i have removed the app from the watch but i can reinstall it to test some things for you if this helps, let me know what you need.

>Also your description of the battery drainage is remarkable. Normally directly or indirectly connecting the watch with the sensor doesn't make a difference. And where do you see that Juggluco uses 33% of the battery budget. When I look in the Wearable app under Watch settings->Battery, Juggluco isn't shown under the apps listen there. Listed are: Display 2.1%, Samsung Heath 1.2%, Health service 0.6%, One Ui Watch Home 0.3%, Google Play Store 0.2%, Watch Faces 0.2%. You have there at the top Juggluco 33.3% ?????? ![battery](https://user-images.githubusercontent.com/72013079/263451387-851bf498-6ee5-477d-97e5-81069627900f.jpg)
> 
> Under Watch settings->Apps->Juggluco is mentioned "_Battery 0% used since last fully charged_". ![appinfo](https://user-images.githubusercontent.com/72013079/263451407-b5fb21e4-efe1-4060-8d5a-f8ed4d36a746.jpg)

thanks for saying this is not normal behaviour, this brings up my hope to get a working solution in the future.
i will conduct additional testing and let you know if i get the same results as mentioned above.

> NFC can't be used any application under WearOS, it is monopolized by Google Pay.

thats a pitty, but something i can imagine google would do ...

**Comment by jrsmile** [comment]
one additional question, if i have direct sensor connection with the watch, does the activated libreview upload on the phone get the values from the watch and uploads them, i ask because i had no values in libreview during the timeframe i switched to the watch. might be impatience on my side though.

**Reply by j-kaltes** [reply]
It doesn't make a difference from what the phone receives the glucose values, it will do the same with them. When it receives the glucose values from the watch (or a mirror connection), the phone will send them to Libreview, to Kerfstok, use them in the xDrip/Nightscout Webserver in Juggluco, the Uploader or Watchdrip if Juggluco is configured as such.

**Comment by j-kaltes** [comment]
> i have removed the app from the watch but i can reinstall it to test some things for you if this helps, let me know what you need.

Because glucose values are send to the phone, it doesn't matter that you uninstalled Juggluco from the watch. Send me the file saved with left middle menu->Export->Stream and tell me when you tried out the WearOS version of Juggluco.

**Reply by jrsmile** [reply]
i have installed 5.1.8 now and test again, i will send you new values as i cant pinpoint when i used the watch the last time.

**Reply by j-kaltes** [reply]
Don't needs to be that specific. If you know the day, it is good enough.

**Reply by jrsmile** [reply]
here you go, switch to watch at 11:34.

SENSORID	12499	1693041849	2023-08-26-11:24:09	2	12498	108	-0.01	STABLE
SENSORID	12500	1693041907	2023-08-26-11:25:07	2	12499	107	-0.07	STABLE
SENSORID	12501	1693041968	2023-08-26-11:26:08	2	12500	105	-0.13	STABLE
SENSORID	12502	1693042029	2023-08-26-11:27:09	2	12501	105	-0.15	STABLE
SENSORID	12503	1693042088	2023-08-26-11:28:08	2	12502	105	-0.14	STABLE
SENSORID	12504	1693042148	2023-08-26-11:29:08	2	12503	106	-0.09	STABLE
SENSORID	12505	1693042208	2023-08-26-11:30:08	2	12504	107	-0.02	STABLE
SENSORID	12506	1693042267	2023-08-26-11:31:07	2	12505	109	+0.06	STABLE
SENSORID	12507	1693042327	2023-08-26-11:32:07	2	12506	110	+0.15	STABLE
SENSORID	12508	1693042388	2023-08-26-11:33:08	2	12507	111	+0.18	STABLE
SENSORID	12509	1693042457	2023-08-26-11:34:17	2	12508	110	+0.14	STABLE
SENSORID	12510	1693042512	2023-08-26-11:35:12	2	12509	107	+nan	NOT_DETERMINED
SENSORID	12511	1693042572	2023-08-26-11:36:12	2	12510	104	-0.16	STABLE
SENSORID	12512	1693042631	2023-08-26-11:37:11	2	12511	104	+nan	NOT_DETERMINED
SENSORID	12513	1693042691	2023-08-26-11:38:11	2	12512	104	-0.2	STABLE
SENSORID	12514	1693042751	2023-08-26-11:39:11	2	12513	106	+nan	NOT_DETERMINED
SENSORID	12515	1693042811	2023-08-26-11:40:11	2	12514	108	-0.01	STABLE
SENSORID	12516	1693042872	2023-08-26-11:41:12	2	12515	108	+nan	NOT_DETERMINED
SENSORID	12517	1693042932	2023-08-26-11:42:12	2	12516	107	-0.01	STABLE
SENSORID	12518	1693042993	2023-08-26-11:43:13	2	12517	105	+nan	NOT_DETERMINED
SENSORID	12519	1693043053	2023-08-26-11:44:13	2	12518	104	-0.13	STABLE
SENSORID	12520	1693043111	2023-08-26-11:45:11	2	12519	104	+nan	NOT_DETERMINED
SENSORID	12521	1693043171	2023-08-26-11:46:11	2	12520	104	-0.15	STABLE
SENSORID	12522	1693043231	2023-08-26-11:47:11	2	12521	105	+nan	NOT_DETERMINED
SENSORID	12523	1693043291	2023-08-26-11:48:11	2	12522	105	-0.1	STABLE
SENSORID	12524	1693043352	2023-08-26-11:49:12	2	12523	106	+nan	NOT_DETERMINED
SENSORID	12525	1693043412	2023-08-26-11:50:12	2	12524	106	-0.08	STABLE
SENSORID	12526	1693043472	2023-08-26-11:51:12	2	12525	106	+nan	NOT_DETERMINED
SENSORID	12527	1693043532	2023-08-26-11:52:12	2	12526	105	-0.06	STABLE

**Reply by j-kaltes** [reply]
You didn't have the previous data because you uninstalled Juggluco and only to test it out you installed it again?
And what does Left menu->Sensor on the watch show?
Did you test it out this short or did you try it somewhat longer?
What happens when you force close the phone version and start it again? (Maybe it was still somehow connected).
Did you give the WearOS version of the watch background permission? (In the Wearable app under app info for Juggluco).
Did you try rebooting the watch?

**Comment by jrsmile** [comment]
> You didn't have the previous data because you uninstalled Juggluco and only to test it out you installed it again? 

exactly

>And what does Left menu->Sensor on the watch show? 

it alternates between last error and last success in the connection row
handshake and glucose value always show correct but every two minutes a one minute delayed timestamp.
Status=133
CHAR_PATCH_CONTROL


>Did you test it out this short or did you try it somewhat longer? 

i am still testing.

> What happens when you force close the phone version and start it again? (Maybe it was still somehow connected). 

the phone version waits for new data, after 2 minutes it syncs with the watch and shows the current value, still only every 2nd datapoint.

> Did you give the WearOS version of the watch background permission? (In the Wearable app under app info for Juggluco). 

yes, rechecked after reboot.

> Did you try rebooting the watch?

yes, it took a few minutes but without intervention or starting anything manual the data came back on watch and phone.
still only every 2 minutes.


here the stream output during app close and watch reboot:

SENSORID	12499	1693041849	2023-08-26-11:24:09	2	12498	108	-0.01	STABLE
SENSORID	12500	1693041907	2023-08-26-11:25:07	2	12499	107	-0.07	STABLE
SENSORID	12501	1693041968	2023-08-26-11:26:08	2	12500	105	-0.13	STABLE
SENSORID	12502	1693042029	2023-08-26-11:27:09	2	12501	105	-0.15	STABLE
SENSORID	12503	1693042088	2023-08-26-11:28:08	2	12502	105	-0.14	STABLE
SENSORID	12504	1693042148	2023-08-26-11:29:08	2	12503	106	-0.09	STABLE
SENSORID	12505	1693042208	2023-08-26-11:30:08	2	12504	107	-0.02	STABLE
SENSORID	12506	1693042267	2023-08-26-11:31:07	2	12505	109	+0.06	STABLE
SENSORID	12507	1693042327	2023-08-26-11:32:07	2	12506	110	+0.15	STABLE
SENSORID	12508	1693042388	2023-08-26-11:33:08	2	12507	111	+0.18	STABLE
SENSORID	12509	1693042457	2023-08-26-11:34:17	2	12508	110	+0.14	STABLE
SENSORID	12510	1693042512	2023-08-26-11:35:12	2	12509	107	+nan	NOT_DETERMINED
SENSORID	12511	1693042572	2023-08-26-11:36:12	2	12510	104	-0.16	STABLE
SENSORID	12512	1693042631	2023-08-26-11:37:11	2	12511	104	+nan	NOT_DETERMINED
SENSORID	12513	1693042691	2023-08-26-11:38:11	2	12512	104	-0.2	STABLE
SENSORID	12514	1693042751	2023-08-26-11:39:11	2	12513	106	+nan	NOT_DETERMINED
SENSORID	12515	1693042811	2023-08-26-11:40:11	2	12514	108	-0.01	STABLE
SENSORID	12516	1693042872	2023-08-26-11:41:12	2	12515	108	+nan	NOT_DETERMINED
SENSORID	12517	1693042932	2023-08-26-11:42:12	2	12516	107	-0.01	STABLE
SENSORID	12518	1693042993	2023-08-26-11:43:13	2	12517	105	+nan	NOT_DETERMINED
SENSORID	12519	1693043053	2023-08-26-11:44:13	2	12518	104	-0.13	STABLE
SENSORID	12520	1693043111	2023-08-26-11:45:11	2	12519	104	+nan	NOT_DETERMINED
SENSORID	12521	1693043171	2023-08-26-11:46:11	2	12520	104	-0.15	STABLE
SENSORID	12522	1693043231	2023-08-26-11:47:11	2	12521	105	+nan	NOT_DETERMINED
SENSORID	12523	1693043291	2023-08-26-11:48:11	2	12522	105	-0.1	STABLE
SENSORID	12524	1693043352	2023-08-26-11:49:12	2	12523	106	+nan	NOT_DETERMINED
SENSORID	12525	1693043412	2023-08-26-11:50:12	2	12524	106	-0.08	STABLE
SENSORID	12526	1693043472	2023-08-26-11:51:12	2	12525	106	+nan	NOT_DETERMINED
SENSORID	12527	1693043532	2023-08-26-11:52:12	2	12526	105	-0.06	STABLE
SENSORID	12528	1693043593	2023-08-26-11:53:13	2	12527	106	+nan	NOT_DETERMINED
SENSORID	12529	1693043651	2023-08-26-11:54:11	2	12528	107	+0	STABLE
SENSORID	12530	1693043713	2023-08-26-11:55:13	2	12529	108	+nan	NOT_DETERMINED
SENSORID	12531	1693043773	2023-08-26-11:56:13	2	12530	110	+nan	NOT_DETERMINED
SENSORID	12532	1693043833	2023-08-26-11:57:13	2	12531	112	+nan	NOT_DETERMINED
SENSORID	12533	1693043893	2023-08-26-11:58:13	2	12532	112	+0.2	STABLE
SENSORID	12534	1693043952	2023-08-26-11:59:12	2	12533	109	+nan	NOT_DETERMINED
SENSORID	12535	1693044012	2023-08-26-12:00:12	2	12534	106	-0.07	STABLE
SENSORID	12536	1693044074	2023-08-26-12:01:14	2	12535	103	+nan	NOT_DETERMINED
SENSORID	12537	1693044134	2023-08-26-12:02:14	2	12536	101	-0.29	STABLE
SENSORID	12538	1693044191	2023-08-26-12:03:11	2	12537	100	+nan	NOT_DETERMINED
SENSORID	12539	1693044251	2023-08-26-12:04:11	2	12538	100	+nan	NOT_DETERMINED
SENSORID	12540	1693044311	2023-08-26-12:05:11	2	12539	101	-0.26	STABLE
SENSORID	12541	1693044370	2023-08-26-12:06:10	2	12540	102	+nan	NOT_DETERMINED
SENSORID	12542	1693044430	2023-08-26-12:07:10	2	12541	104	-0.09	STABLE
SENSORID	12543	1693044491	2023-08-26-12:08:11	2	12542	102	+nan	NOT_DETERMINED
SENSORID	12544	1693044551	2023-08-26-12:09:11	2	12543	101	+nan	NOT_DETERMINED
SENSORID	12545	1693044611	2023-08-26-12:10:11	2	12544	102	-0.07	STABLE
SENSORID	12546	1693044673	2023-08-26-12:11:13	2	12545	103	+nan	NOT_DETERMINED
SENSORID	12547	1693044733	2023-08-26-12:12:13	2	12546	104	+nan	NOT_DETERMINED
SENSORID	12548	1693044793	2023-08-26-12:13:13	2	12547	104	+0	STABLE
SENSORID	12549	1693044851	2023-08-26-12:14:11	2	12548	104	+nan	NOT_DETERMINED
SENSORID	12550	1693044911	2023-08-26-12:15:11	2	12549	103	-0.07	STABLE
SENSORID	12551	1693044971	2023-08-26-12:16:11	2	12550	103	+nan	NOT_DETERMINED
SENSORID	12552	1693045031	2023-08-26-12:17:11	2	12551	102	-0.13	STABLE
SENSORID	12553	1693045094	2023-08-26-12:18:14	2	12552	102	+nan	NOT_DETERMINED
SENSORID	12554	1693045154	2023-08-26-12:19:14	2	12553	101	-0.16	STABLE
SENSORID	12555	1693045213	2023-08-26-12:20:13	2	12554	101	+nan	NOT_DETERMINED
SENSORID	12556	1693045273	2023-08-26-12:21:13	2	12555	100	-0.17	STABLE
SENSORID	12557	1693045331	2023-08-26-12:22:11	2	12556	100	+nan	NOT_DETERMINED
SENSORID	12558	1693045391	2023-08-26-12:23:11	2	12557	101	-0.11	STABLE
SENSORID	12559	1693045452	2023-08-26-12:24:12	2	12558	102	+nan	NOT_DETERMINED
SENSORID	12560	1693045512	2023-08-26-12:25:12	2	12559	104	+0.02	STABLE
SENSORID	12561	1693045571	2023-08-26-12:26:11	2	12560	105	+nan	NOT_DETERMINED
SENSORID	12562	1693045631	2023-08-26-12:27:11	2	12561	106	+0.11	STABLE
SENSORID	12563	1693045691	2023-08-26-12:28:11	2	12562	106	+nan	NOT_DETERMINED
SENSORID	12564	1693045751	2023-08-26-12:29:11	2	12563	107	+0.16	STABLE
SENSORID	12565	1693045813	2023-08-26-12:30:13	2	12564	108	+nan	NOT_DETERMINED
SENSORID	12566	1693045873	2023-08-26-12:31:13	2	12565	108	+0.18	STABLE
SENSORID	12567	1693045931	2023-08-26-12:32:11	2	12566	109	+nan	NOT_DETERMINED
SENSORID	12568	1693045991	2023-08-26-12:33:11	2	12567	110	+0.21	STABLE
SENSORID	12569	1693046052	2023-08-26-12:34:12	2	12568	110	+nan	NOT_DETERMINED
SENSORID	12570	1693046112	2023-08-26-12:35:12	2	12569	110	+nan	NOT_DETERMINED
SENSORID	12571	1693046172	2023-08-26-12:36:12	2	12570	111	+0.22	STABLE
SENSORID	12572	1693046230	2023-08-26-12:37:10	2	12571	112	+nan	NOT_DETERMINED
SENSORID	12573	1693046290	2023-08-26-12:38:10	2	12572	113	+0.29	STABLE
SENSORID	12574	1693046350	2023-08-26-12:39:10	2	12573	113	+nan	NOT_DETERMINED
SENSORID	12575	1693046410	2023-08-26-12:40:10	2	12574	114	+0.27	STABLE
SENSORID	12576	1693046471	2023-08-26-12:41:11	2	12575	114	+nan	NOT_DETERMINED
SENSORID	12577	1693046531	2023-08-26-12:42:11	2	12576	114	+0.29	STABLE
SENSORID	12578	1693046590	2023-08-26-12:43:10	2	12577	115	+nan	NOT_DETERMINED
SENSORID	12579	1693046650	2023-08-26-12:44:10	2	12578	116	+0.36	STABLE
SENSORID	12580	1693046712	2023-08-26-12:45:12	2	12579	118	+nan	NOT_DETERMINED
SENSORID	12581	1693046772	2023-08-26-12:46:12	2	12580	118	+nan	NOT_DETERMINED
SENSORID	12582	1693046832	2023-08-26-12:47:12	2	12581	118	+0.38	STABLE
SENSORID	12583	1693046890	2023-08-26-12:48:10	2	12582	117	+nan	NOT_DETERMINED
SENSORID	12584	1693046950	2023-08-26-12:49:10	2	12583	117	+0.32	STABLE
SENSORID	12585	1693047011	2023-08-26-12:50:11	2	12584	116	+nan	NOT_DETERMINED
SENSORID	12586	1693047071	2023-08-26-12:51:11	2	12585	116	+nan	NOT_DETERMINED
SENSORID	12587	1693047131	2023-08-26-12:52:11	2	12586	116	+nan	NOT_DETERMINED
SENSORID	12588	1693047191	2023-08-26-12:53:11	2	12587	115	+nan	NOT_DETERMINED
SENSORID	12589	1693047251	2023-08-26-12:54:11	2	12588	115	+0.14	STABLE
SENSORID	12590	1693047308	2023-08-26-12:55:08	2	12589	114	+nan	NOT_DETERMINED
SENSORID	12591	1693047368	2023-08-26-12:56:08	2	12590	113	+0.06	STABLE
SENSORID	12593	1693047491	2023-08-26-12:58:11	2	12592	116	+0.2	STABLE

**Reply by j-kaltes** [reply]
Because your are not a user of Juggluco, you used another app or device to connect with the Libre 3 sensor. Are you certain that Abbott’s Libre 3 app or reader wasn’t active in the background?

**Reply by jrsmile** [reply]
certanly, i never used the abbott reader. but i can shutdown my phone which is the only device i ever used with the sensor and observe the stream on the watch. give me ten minutes.

**Reply by j-kaltes** [reply]
Yes, I also shouldn't have mentioned Abbott's Libre 3 reader, because Juggluco can't take over Libre 3 sensors from it because it's Libreview Account ID is unknown.

**Comment by jrsmile** [comment]
bad news, i haven't even started my phone yet to get the stream export, but Left Menu>Sensor still shows bi minute errors.
after starting the phone i got the following stream log:
SENSORID	12593	1693047491	2023-08-26-12:58:11	2	12592	116	+0.2	STABLE
SENSORID	12594	1693047549	2023-08-26-12:59:09	2	12593	118	+nan	NOT_DETERMINED
SENSORID	12595	1693047609	2023-08-26-13:00:09	2	12594	115	+0.14	STABLE
SENSORID	12596	1693047668	2023-08-26-13:01:08	2	12595	109	+nan	NOT_DETERMINED
SENSORID	12597	1693047728	2023-08-26-13:02:08	2	12596	102	-0.44	STABLE
SENSORID	12598	1693047789	2023-08-26-13:03:09	2	12597	96	+nan	NOT_DETERMINED
SENSORID	12599	1693047849	2023-08-26-13:04:09	2	12598	91	-0.93	STABLE
SENSORID	12600	1693047908	2023-08-26-13:05:08	2	12599	87	+nan	NOT_DETERMINED
SENSORID	12601	1693047968	2023-08-26-13:06:08	2	12600	85	-1.22	FALLING
SENSORID	12602	1693048029	2023-08-26-13:07:09	2	12601	83	+nan	NOT_DETERMINED
SENSORID	12603	1693048089	2023-08-26-13:08:09	2	12602	79	-1.29	FALLING
SENSORID	12604	1693048152	2023-08-26-13:09:12	2	12603	79	+nan	NOT_DETERMINED
SENSORID	12605	1693048212	2023-08-26-13:10:12	2	12604	83	+nan	NOT_DETERMINED
SENSORID	12606	1693048272	2023-08-26-13:11:12	2	12605	90	+nan	NOT_DETERMINED
SENSORID	12607	1693048332	2023-08-26-13:12:12	2	12606	97	-0.03	STABLE
SENSORID	12608	1693048390	2023-08-26-13:13:10	2	12607	102	+nan	NOT_DETERMINED
SENSORID	12609	1693048450	2023-08-26-13:14:10	2	12608	105	+nan	NOT_DETERMINED
SENSORID	12610	1693048510	2023-08-26-13:15:10	2	12609	108	+0	STABLE
SENSORID	12611	1693048570	2023-08-26-13:16:10	2	12610	110	+nan	NOT_DETERMINED
SENSORID	12612	1693048630	2023-08-26-13:17:10	2	12611	112	+nan	NOT_DETERMINED
SENSORID	12613	1693048690	2023-08-26-13:18:10	2	12612	112	+0.32	STABLE
SENSORID	12614	1693048749	2023-08-26-13:19:09	2	12613	112	+nan	NOT_DETERMINED
SENSORID	12615	1693048809	2023-08-26-13:20:09	2	12614	111	+0.1	STABLE
SENSORID	12616	1693048872	2023-08-26-13:21:12	2	12615	107	+nan	NOT_DETERMINED
SENSORID	12617	1693048932	2023-08-26-13:22:12	2	12616	104	-0.13	STABLE
SENSORID	12618	1693048989	2023-08-26-13:23:09	2	12617	107	+nan	NOT_DETERMINED
SENSORID	12619	1693049049	2023-08-26-13:24:09	2	12618	108	-0.01	STABLE
SENSORID	12620	1693049108	2023-08-26-13:25:08	2	12619	107	+nan	NOT_DETERMINED
SENSORID	12621	1693049168	2023-08-26-13:26:08	2	12620	106	-0.07	STABLE
SENSORID	12622	1693049229	2023-08-26-13:27:09	2	12621	104	+nan	NOT_DETERMINED
SENSORID	12623	1693049289	2023-08-26-13:28:09	2	12622	103	-0.14	STABLE
SENSORID	12624	1693049348	2023-08-26-13:29:08	2	12623	102	+nan	NOT_DETERMINED
SENSORID	12625	1693049408	2023-08-26-13:30:08	2	12624	102	-0.12	STABLE
SENSORID	12626	1693049470	2023-08-26-13:31:10	2	12625	102	+nan	NOT_DETERMINED
SENSORID	12627	1693049530	2023-08-26-13:32:10	2	12626	103	+nan	NOT_DETERMINED
SENSORID	12628	1693049590	2023-08-26-13:33:10	2	12627	103	-0.02	STABLE
SENSORID	12629	1693049648	2023-08-26-13:34:08	2	12628	101	+nan	NOT_DETERMINED
SENSORID	12630	1693049708	2023-08-26-13:35:08	2	12629	99	-0.14	STABLE
SENSORID	12631	1693049770	2023-08-26-13:36:10	2	12630	99	+nan	NOT_DETERMINED
SENSORID	12632	1693049830	2023-08-26-13:37:10	2	12631	99	-0.23	STABLE
SENSORID	12633	1693049890	2023-08-26-13:38:10	2	12632	94	+nan	NOT_DETERMINED
SENSORID	12634	1693049950	2023-08-26-13:39:10	2	12633	92	-0.47	STABLE
SENSORID	12635	1693050008	2023-08-26-13:40:08	2	12634	90	+nan	NOT_DETERMINED
SENSORID	12636	1693050068	2023-08-26-13:41:08	2	12635	89	-0.51	STABLE
SENSORID	12637	1693050130	2023-08-26-13:42:10	2	12636	89	+nan	NOT_DETERMINED
SENSORID	12638	1693050190	2023-08-26-13:43:10	2	12637	91	-0.38	STABLE
SENSORID	12639	1693050251	2023-08-26-13:44:11	2	12638	93	+nan	NOT_DETERMINED
SENSORID	12640	1693050311	2023-08-26-13:45:11	2	12639	95	-0.13	STABLE
SENSORID	12641	1693050372	2023-08-26-13:46:12	2	12640	96	+nan	NOT_DETERMINED
SENSORID	12642	1693050432	2023-08-26-13:47:12	2	12641	96	+nan	NOT_DETERMINED
SENSORID	12643	1693050492	2023-08-26-13:48:12	2	12642	94	+nan	NOT_DETERMINED
SENSORID	12644	1693050552	2023-08-26-13:49:12	2	12643	92	-0.27	STABLE

**Reply by j-kaltes** [reply]
Can you show the content of Left menu->Sensor on the watch in a few screenshots? (press both buttons).

**Reply by j-kaltes** [reply]
Now I see the SensorID, you had edited out of the exported Stream data.
Status=0? Did you turn turn Bluetooth off and on? Do you have status=0 more often?
Did you try turning on Turbo?

**Comment by jrsmile** [comment]
yea, im not very fond to have the sensorid on the web but if its necessary.
it is more often Status=133 but ocasionly Status=0 but i never turned of bluetooth manually.
i have enabled turbo now, lets see if there is a difference.

**Reply by j-kaltes** [reply]
I didn't need _Turbo_ for Libre3, but for Libre2 had maybe made a difference.

**Reply by jrsmile** [reply]
i can enable wakelock and android 13 if you want to test that too.

**Reply by jrsmile** [reply]
with turbo still only every 2 minutes.

**Reply by j-kaltes** [reply]
Wakelock and Android 13 are certainly not necessary. I think about removing these options; to many people turn them on.

**Comment by jrsmile** [comment]
i can provide you with a teamviewer session and android studio to be able to debug directly via adb if you want that?

**Reply by j-kaltes** [reply]
You can install a logging version of Juggluco on your watch and send the logcat output to me. For this you can as well use 5.1.3 from https://www.juggluco.nl/Juggluco/download.html 
See there under "BUG Report".

**Reply by jrsmile** [reply]
do i have to install the logcat version on the phone also? i ask because under left menu> watch>wear os> config  there is no longer my watch listed, so i can't init app and move sensor to watch. current config: phone 5.1.8 watch 5.1.3-wear

**Reply by j-kaltes** [reply]
Does the Wearable app show a connection?

**Reply by jrsmile** [reply]
in the clone menu yes, in the default screen "network problem?"

**Reply by j-kaltes** [reply]
What does the Android Samsung Galaxy Wearable app say?

**Reply by jrsmile** [reply]
nevermind, it took some time, i toggeled bluetooth two times now i could init the app. i moved the sensor to the watch and redisabled battery optimisation. collecting data now.

**Comment by jrsmile** [comment]
i have send the logcat dump to your gmail address.

**Reply by j-kaltes** [reply]
The pattern seems to be repeatedly: getting a connection with the sensor, receiving a glucose value and immediately thereafter an onConnectionStateChange, disconnecting with status=0. Hereafter it takes some time before a connection is re-established.
Status=0 is very unusual. I only see it when I turn Bluetooth off and on and on Android 4.4 or when the app itself disconnects: https://issuetracker.google.com/issues/207323675#comment3
I went through all mentions of disconnect() in Juggluco and none of the associated log messages were logged. The code is the same in the phone version and it doesn’t have these problems.
Something seems wrong with the watch. Did you install some apps doing something with Bluetooth?
You can try to uninstall unused apps: https://forum.xda-developers.com/t/galaxy-watch-6-adb-debloat-connecting.4613093/
I would try out whether a factory reset makes a difference.
EDIT:
Did you already try out settings '_Android 13_'? This changes the way connecting and disconnecting is caried out.

**Reply by jrsmile** [reply]
hi thanks for checking, the watch is basicly stock, i added glucodatahandler and juggluco, there is a lot of samsung bloat on the watch but im not yet commiting to remove it. ive read about an android 13 issue here: https://issuetracker.google.com/issues/242755161?pli=1 which might be of interrest. i activated the android13 option an currently test reliability.

**Reply by jrsmile** [reply]
i tried android 13 as option, it behaves "differently" at first it took longer to get a value (3-5 min) toggeling bluetooth helped. i then added the juggluco watchface, next step was to turn "always on display" on. to give the app as much "priority" as possible, then i went into the clone menu and switched on wifi to test if the connection with the phone disables ble somehow. all actions had no effect. all in all i would say the android 13 option does work less reliable.

**Reply by jrsmile** [reply]
Good Day, i have removed most of the Stock apps on the watch following the link you posted. On watch 6 there where a buch more default apps some directly indicating working with bluetooth. Like galaxy buds controller. I removed them as well. Unfortunately there Was no change in the behaviour. I rebooted watch and phone after uninstallation of the bloatware.

**Comment by Puderzo** [comment]
Did you ever figure out what's causing this?

We noticed the same behavior when switching my son's watch from Galaxy Watch 4 Classic to Galaxy Watch 6 Classic a few months ago; instead of getting reliable 1-minute readings, we're getting readings only once every few minutes. This hasn't been a huge issue for us (we export to xDrip and that's what we parents then follow, and that only updates every 5 minutes anyway) but sometimes when the BG is acting up I switch back to the phone as collector, just to make sure I get all the readings.

Perhaps this is related to WearOS 4, as the Watch 6 came with that out of the box, whereas Watch 4 was still running 3.5 when we switched? Maybe some "optimization" is trying to save battery this way (in my experience, the battery lasts about as long on GW6 as on GW4, which is at most about a day, so whatever it tries to do doesn't work as intended).
